### PR TITLE
Drag and drop

### PR DESCRIPTION
**Pull request by Mimiste**
Possibility to initiate a drag and drop

- Touch for 2 second to initiate a drag
- Move the cursor as usual
- Tap to drop

### CODE REVIEW

File: Limelight/Input/StreamView.m [hunk 1]
```diff
@@ -49,11 +52,25 @@ - (void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event {
         UITouch *touch = [[event allTouches] anyObject];
         touchLocation = [touch locationInView:self];
         touchMoved = false;
+        dragTimer = [NSTimer scheduledTimerWithTimeInterval:2.0
```
Comment (cgutman) [inline Limelight/Input/StreamView.m]: Can we get another touchesBegan before touchesEnded? We don't want to leave a timer dangling.

File: Limelight/Input/StreamView.m [hunk 2]
```diff
@@ -49,11 +52,25 @@ - (void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event {
...
+                                                     target:self
+                                                   selector:@selector(onDragStart:)
+                                                   userInfo:nil
+                                                    repeats:NO];
+    }
+}
+
+- (void)onDragStart:(NSTimer*)timer {
+    if (!touchMoved && !isDragging){
```
Comment (cgutman) [inline Limelight/Input/StreamView.m]: It looks like if this code fires for a 2 finger drag, the left button will never be released.

Comment (Mimiste) [inline Limelight/Input/StreamView.m]: I corrected this part to start the timer only if you touch with one finger.

But, do you have something for testing the app ? 
My problem is each time i want to test something, i have to reboot the computer under windows to launch geforce experience..

**Comment by dwaxemberg** [comment]
I tried this out and it works pretty well, but 2 seconds is a very long time to not move your finger. Can we reduce that to maybe 1 second?

**Comment by Mimiste** [comment]
You are right. After 1 week of using it's clearly too long. I changed it to 1 second

**Pull request merged**